Documentation/app-operations.md

```diff
@@ -0,0 +1,232 @@
+3. The entrypoint will be executed and passed a reference of the application that was prepared.
+
+The responsibility of this entrypoint is to receive a prepared app, inject it into the pod, 
+so that it is ready to be started by the stage1 (e.g. systemd for the default flavor).
+
+### Stage1 Default Flavor Example Workflow
+
+1. Render the application rootfs.
+2. Prepare the application/s
+3. Start the injected application using the `app start` entrypoint.
+
+This approach is similar to using `rkt run`, instead of creating a new pod,
+an existing pod will be used.
+
+## rkt app rm
+
+`coreos.com/rkt/stage1/app/rm`
+
+1. receive a reference to an application that resides inside the pod (running or stopped)
+2. stop the application if its running.
+3. remove the contents of the application (rootfs) from the pod (keep the logs?) and delete references to it (e.g. service files).
+   - Delegate these steps by creating `app/install` and `app/rm` entrypoints?
+
+The responsibility of this entrypoint is to receive an app inside a pod and to remove it from the pod, so that it is ready to be started by the stage1 (e.g. systemd for the default flavor).
+After `rm`, starting the application again is not possible and requires injecting the same
+application using the `app/add` entrypoint.
+
+### Stage1 Default Flavor Example Workflow
+
+1. Stop the application using the `app stop` entrypoint.
+2. Collect the garbage left by the app (e.g. rootfs, unit files, etc.)
+3. Reload the systemd daemon for changes to take place without disturbing the other tennant applications.
+
+## rkt app start
+
+`coreos.com/rkt/stage1/app/start`
+
+1. given a reference to an application that's in a startable state (see below), instruct the to start the application.
+
+The responsibility of this entrypoint is to start an application that is in a `Ready` or `Exited` state, 
+which is an app that was recently injected, instruct the stage1 to start the application.
+
+## rkt app stop
+
+`coreos.com/rkt/stage1/app/stop`
+
+1. given a reference to an application that's in a stoppable state (see below), instruct the stage1 to stop the application.
+
+The responsibility of this entrypoint is to stop an application that is `Running` by instructing the stage1. 
+
+**Note:** there is a graceful shutdown which sends the termination signal to application and waits for a grace period
+for application to terminate and then the application is shutdown if it doesn't terminate by the end of the grace period.
+
+# App States
+
+Expected set of app states are listed below:
+
+```go
+type AppState string
+
+const (
+	UnknownAppState AppState = "unknown"
+
+	PreparingAppState AppState = "preparing"
+
+	// Apps that are ready to be used by `app start`.
+	PreparedAppState AppState = "prepared"
+
+	RunningAppState AppState = "running"
+
+	// Apps stopped by `app stop`.
+	StoppingAppState AppState = "stopping"
+
+	// Apps that finish their execution naturally.
```

Comment:
Will the transition to the exited state happen from "running" or from "stopping" ?